catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -204,6 +212,26 @@ QueryResponse doQuery(QueryRequest queryRequest, FederationStrategy strategy)
 
         LOGGER.debug("Calling strategy.federate()");
 
+        Query originalQuery = queryRequest.getQuery();
+
+        long timeout = originalQuery.getTimeoutMillis();
+
+        if (timeout == 0) {
+            timeout = queryTimeoutMillis;
```

Comment:
The goal was to be able to control a maximum timeout if an unlimited timeout was chosen.  We do not want to bugs to allow queries to stay open forever and allow leaks.  There should still be a maximum timeout for all queries in the system.